### DISCUSSION TITLE
Multiple Y-axes for a plot

### DISCUSSION THREAD

**Discussion by vit0l** [pyqtgraph/pyqtgraph#2031] [General]
For my application I need to plot several data-sets which have the same X-axis (time) and different Y-axes (units as well as scaling).
Is it possible to have multiple independent Y-axes for the same plot (the same X-axis)?

**Comment by j9ac9k** [comment]
Hi @vit0l 

This is a frequently requested feature, and it can be done in the library as is, but I'll be the first to admit it's a painful process.

Fortunately, there are 2 PRs that are attempting to roll out this functionality, #2010 and #1359  If you could give either/both a try and report how well they work for you, that would be really helpful, as we would like to merge this functionality within the library itself.

**Comment by vit0l** [comment]
Ok, thanks! I'll try and share here an output.

**Comment by danielhrisca** [comment]
I've also implemented a Plot wrapper that can handle multiple Y axis. You can check it out [here](https://github.com/danielhrisca/asammdf/blob/1507f100bddc86c5f037db2770ec39e42bdfc306/asammdf/gui/widgets/plot.py#L3576)

**Reply by j9ac9k** [reply]
FWIW we're working on this feature in #1359 right now... we're still seeing some weird edge cases so hopefully soon we'll have a built-in method to handle multiple axes on the same plot.

**Comment by MaxHaufe** [comment]
Hi, I also need this feature my bachelor's thesis. I have checked out the work of 0000matteo0000 from #1359 and I really like it. 

Will this be merged anytime in the near future or am I just better off using matteos branch even though I am afraid that I will miss out on future updates if it will not be synced with pyqtgraph:master?
